cdc/puller/entry_sorter.go

```diff
@@ -0,0 +1,121 @@
+package puller
+
+import (
+	"context"
+	"sort"
+	"sync"
+
+	"github.com/pingcap/ticdc/cdc/model"
+)
+
+// EntrySorter is used to sort entries outputted from puller
+type EntrySorter struct {
+	unsorted   []*model.RawKVEntry
+	resolvedCh chan uint64
+	lock       sync.Mutex
+
+	output chan *model.RawKVEntry
+}
+
+// NewEntrySorter creates a new EntrySorter
+func NewEntrySorter() *EntrySorter {
+	return &EntrySorter{
+		resolvedCh: make(chan uint64, 1024),
+		output:     make(chan *model.RawKVEntry, 128),
+	}
+}
+
+// Run runs the EntrySorter, sort the entries
+func (es *EntrySorter) Run(ctx context.Context) {
+	lessFunc := func(i *model.RawKVEntry, j *model.RawKVEntry) bool {
+		if i.Ts == j.Ts {
+			return i.OpType == model.OpTypeDelete && j.OpType == model.OpTypePut
```

Comment:
```suggestion
			return i.OpType == model.OpTypeDelete
```
only  check`entry-i` is `delete` can save one comparison, and meet our requirement: `always put Delete event before Put event`. But not accurate `less than` semantic